### PR TITLE
OPML Export

### PR DESCRIPTION
**Pull request by fabrixxm**
Export user's active RSS/Atom contacts as OPML.

### CODE REVIEW

File: opmlexport/opmlexport.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+use Friendica\DI;
+use Friendica\App;
+use Friendica\Core\Hook;
+use Friendica\Core\Logger;
+use Friendica\Network\HTTPException;
+use Friendica\Database\DBA;
+use Friendica\Core\Renderer;
+
+function opmlexport_install() {
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]: Code standards: Please add a new line before function opening braces.
```suggestion
function opmlexport_install()
{
```

File: opmlexport/opmlexport.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    Hook::register("addon_settings",        __FILE__, "opmlexport_addon_settings");
+    Hook::register("addon_settings_post",   __FILE__, "opmlexport_addon_settings_post");
+    Logger::log("installed opmlexport Addon");
+}
+
+
+function opmlexport(App $a) {
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]:
```suggestion
function opmlexport(App $a)
{
```

File: opmlexport/opmlexport.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    $stmt = DBA::p("SELECT *
+            FROM `contact`
+            WHERE `uid` = ?
+                AND `self` = 0
+                AND NOT `deleted`
+                AND NOT `archive` AND NOT `blocked` AND NOT `pending`
+                AND network = ?
+            ORDER BY `name` ASC",
+            [local_user(), "feed"]
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]: Please use the corresponding constant: `\Friendica\Core\Protocol::FEED`
```suggestion
            [local_user(), \Friendica\Core\Protocol::FEED]
```

File: opmlexport/opmlexport.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    $outline = $xml->createElement("outline");
+    $outline->setAttribute("title", $a->user['username'] . "'s RSS/Atom contacts");
+    $outline->setAttribute("text", $a->user['username'] . "'s RSS/Atom contacts");
+
+    while ($c = DBA::fetch($stmt)) {
+        $entry = $xml->createElement("outline");
+        $entry->setAttribute("title",  $c["name"]);
+        $entry->setAttribute("text",   $c["name"]);
+        $entry->setAttribute("xmlUrl", $c["url"]);
+        // $entry->setAttribute("htmlUrl", $c[""]);
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]: Please remove commented out code.

File: opmlexport/opmlexport.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    $xml->appendChild($opml);
+    header("Content-Type: text/x-opml");
+    header("Content-Disposition: attachment; filename=feeds.opml");
+    $xml->formatOutput = true;
+    echo $xml->saveXML();
+    die();
+}
+
+
+function opmlexport_addon_settings(App $a, &$s) {
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]:
```suggestion
function opmlexport_addon_settings(App $a, &$s)
{
```

File: opmlexport/opmlexport.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                '$submit'  => DI::l10n()->t('Export RSS/Atom contacts'),
+        ]);
+}
+
+
+function opmlexport_addon_settings_post(App $a, &$b)
+{
+        if (!local_user() || empty($_POST['opmlexport-submit'])) {
+                return;
+        }
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]: Code standards: Please make sure the indentation is tabs throughout the whole file.
```suggestion
	if (!local_user() || empty($_POST['opmlexport-submit'])) {
		return;
	}
```

File: opmlexport/opmlexport.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+use Friendica\Core\Hook;
+use Friendica\Core\Logger;
+use Friendica\Network\HTTPException;
+use Friendica\Database\DBA;
+use Friendica\Core\Renderer;
+
+function opmlexport_install() {
+    Hook::register("addon_settings",        __FILE__, "opmlexport_addon_settings");
+    Hook::register("addon_settings_post",   __FILE__, "opmlexport_addon_settings_post");
+    Logger::log("installed opmlexport Addon");
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]: Code standards: Please normalize quotes throughout the file, preferably single quotes when double quotes aren't needed.
```suggestion
    Hook::register('addon_settings',        __FILE__, 'opmlexport_addon_settings');
    Hook::register('addon_settings_post',   __FILE__, 'opmlexport_addon_settings_post');
    Logger::log('installed opmlexport Addon");
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: opmlexport/opmlexport.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+
+function opmlexport(App $a) {
+    $stmt = DBA::p("SELECT *
```
Comment (annando) [inline opmlexport/opmlexport.php]: Could you use this here instead?
```
$condition = ['uid' => local_user(), 'self' => false, 'deleted' => false, 'archive' => false,
	'blocked' => false, 'pending' => false, 'network' => Protocol::FEED];
$data = Contact::selectToArray([], $condition, ['order' => ['name']]);
```

File: opmlexport/opmlexport.php [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    ]);
+}
+
+
+function opmlexport_addon_settings_post(App $a, &$b)
+{
+    if (!local_user() || empty($_POST['opmlexport-submit'])) {
+        return;
+    }
+    opmlexport($a);
```
Comment (MrPetovan) [inline opmlexport/opmlexport.php]: Great, you removed the only correctly indenting tab in the file!

Please replace all the indenting spaces with tabs.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: opmlexport/opmlexport.php (hunk 9 above)
Comment (fabrixxm) [reply opmlexport/opmlexport.php]: nuoooooooooo fkng vscode! aaaargh!

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**